onnxruntime/core/providers/nuphar/scripts/model_editor.py

```diff
@@ -143,6 +150,97 @@ def _bidirectional(outputs, axis, hook_output_name):
         for i_o in range(1, len(node.output)):
             nf.make_node('Unsqueeze', state_outputs[i_o - 1], {'axes':[0]}, output_names=node.output[i_o])
 
+def convert_loop_to_scan(node, out_main_graph):
+
+    initial_state_names = node.input[2:]   # exclude M and cond
+
+    loop_subgraph_input_i = node.attribute[0].g.input[0]
+    subgraph_input_names = []
+    scan_input_names = []
+
+    # 1. find Gather with i as input, Gather.input[0] to be Scan op's scaninputs
+    #   Gather ops are to be removed from the subgraph
```

Comment:
Loop-to-Scan only works for a specific pattern, please add checks to make sure loops don't have those pattern are not converted, instead of crashing.